lib/rules/prefer-destructuring.js

```diff
@@ -110,22 +167,21 @@ module.exports = {
             }
 
             if (isArrayIndexAccess(rightNode)) {
-                if (checkArrays) {
+                if (shouldCheck("VariableDeclarator", "array") || shouldCheck("AssignmentExpression", "array") || checkArrays) {
```

Comment:
Good point. With this configuration, both of these are valid cases, along with a few more cases that I added